internal/compiler-interface/src/main/java/xsbti/BasicVirtualFileRef.java

```diff
@@ -11,6 +11,9 @@
 
 package xsbti;
 
+import java.util.Arrays;
+import java.util.ArrayList;
+
 public class BasicVirtualFileRef implements VirtualFileRef {
     private String parent;
```

Comment:
Could `parent` and `name` be marked final and possibly annotated `@NotNull`? Below it isn't obvious to me in the names method that a null check isn't necessary on parent.

Reply:
Sounds like a good idea.

Reply:
Is there a NotNull that's standard in JDK 8 and 11?

Reply:
Oh, good point. There might not be.